### PR TITLE
Remove unused support for /persist/config

### PR DESCRIPTION
**Pull request by eriknordmark**
This has not been used for a while so remove the remaining code from pubsub package.

### CODE REVIEW

File: pkg/pillar/types/locationconsts.go [hunk 1]
```diff
@@ -13,7 +13,7 @@ const (
 	PersistDir = "/persist"
 	// PersistConfigDir is where we keep some configuration across reboots
 	PersistConfigDir = PersistDir + "/config"
-	// PersistStatusDir is where we keep some configuration across reboots
+	// PersistStatusDir is where we used to keep some configuration across reboots. Remove once upgradeconverter code is removed.
```
Comment (OhmSpectator) [inline pkg/pillar/types/locationconsts.go:16]: I guess it should be a comment to the `PersistConfigDir` constant, not `PersistStatusDir`?

File: pkg/pillar/pubsub/socketdriver/driver.go [hunk 2]
```diff
@@ -87,8 +81,8 @@ func (s *SocketDriver) Publisher(global bool, name, topic string, persistent boo
 	// the dirName depends on if we are persistent, and if it is the global config
 	switch {
 	case persistent && publishToDir:
-		// Special case for /persist/config/
-		dirName = fmt.Sprintf("%s/%s/%s", s.RootDir, persistConfigDir, name)
+		// No longer supported
+		return nil, errors.New("Persistent not supported for empty agentname")
```
Comment (OhmSpectator) [inline pkg/pillar/pubsub/socketdriver/driver.go:85]: I'm trying to understand what are the "global" subscriptions.
I've tried to find any "anonymous" (without agentname) Publisher for example but I don't see any...

File: pkg/pillar/types/locationconsts.go (hunk 1 above)
Comment (eriknordmark) [reply pkg/pillar/types/locationconsts.go:16]: Oops

File: pkg/pillar/pubsub/socketdriver/driver.go (hunk 2 above)
Comment (eriknordmark) [reply pkg/pillar/pubsub/socketdriver/driver.go:85]: I see two on a running system by looking in /run/global/:
DevicePortConfig  LedBlinkCounter

For LedBlinkCounter there is only a subscriber I think - some old legacy code to push changes to that config.

Comment (deitch) [reply pkg/pillar/pubsub/socketdriver/driver.go:85]: Some of it is documented [here](https://github.com/lf-edge/eve/blob/master/docs/IPC.md) @OhmSpectator

Comment (OhmSpectator) [reply pkg/pillar/pubsub/socketdriver/driver.go:85]: @deitch, oh, thanks! I wish I had found this doc much earlier.

File: pkg/pillar/types/locationconsts.go (hunk 1 above)
Comment (eriknordmark) [reply pkg/pillar/types/locationconsts.go]: Fixed

**Pull request merged**